LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -306,9 +342,20 @@ public async Task<bool> SaveFrameCountChangesAsync()
         /// </summary>
         public void AcceptFrameCountChanges()
         {
-            lock (this.fcntLock)
+            this.AcceptFrameCountChanges(this.fcntUp, this.fcntDown);
+        }
+
+        /// <summary>
+        /// Accept changes to the frame count
+        /// </summary>
+        void AcceptFrameCountChanges(int savedFcntUp, int savedFcntDown)
+        {
+            lock (this.fcntSync)
             {
-                this.hasFrameCountChanges = false;
+                this.lastSavedFcntUp = savedFcntUp;
+                this.lastSavedFcntDown = savedFcntDown;
+
+                this.hasFrameCountChanges = this.fcntDown != this.lastSavedFcntDown || this.fcntUp != this.lastSavedFcntUp;
```

Comment:
Nice. There was a race here before that I also fixed in our feature branch. Will have to make sure we get all of your changes. :)